src/System.Security.Cryptography.Csp/tests/PasswordDeriveBytesTests.cs

```diff
@@ -0,0 +1,452 @@
+                ByteUtils.HexToByteArray("B0695D8D98F5844B9650A9F68EFF105B"));
+        }
+
+        [Fact]
+        public static void CryptDeriveKey_KnownValues_DES()
+        {
+            byte[] key = TestKnownValue_CryptDeriveKey(
+                TestPassword,
+                "DES",
+                "SHA1",
+                64,
+                s_testSalt,
+                ByteUtils.HexToByteArray("B0685D8C98F4854A"));
+        }
+
+        [Fact]
+        public static void CryptDeriveKey_Invalid_KeyLength()
+        {
+            using (var deriveBytes = new PasswordDeriveBytes(TestPassword, s_testSalt))
+            {
+                Assert.ThrowsAny<CryptographicException>(() => deriveBytes.CryptDeriveKey("RC2", "SHA1", 127, s_testSalt));
+                Assert.ThrowsAny<CryptographicException>(() => deriveBytes.CryptDeriveKey("RC2", "SHA1", 129, s_testSalt));
+            }
+        }
+
+        [Fact]
+        public static void CryptDeriveKey_Invalid_Algorithm()
+        {
+            using (var deriveBytes = new PasswordDeriveBytes(TestPassword, s_testSalt))
+            {
+                Assert.Throws<CryptographicException>(() => deriveBytes.CryptDeriveKey("BADALG", "SHA1", 128, s_testSalt));
+            }
+        }
+
+        [Fact]
+        public static void CryptDeriveKey_Invalid_HashAlgorithm()
+        {
+            using (var deriveBytes = new PasswordDeriveBytes(TestPassword, s_testSalt))
+            {
+                Assert.Throws<CryptographicException>(() => deriveBytes.CryptDeriveKey("RC2", "BADALG", 128, s_testSalt));
+            }
+        }
+
+        [Fact]
+        public static void CryptDeriveKey_Invalid_IV()
+        {
+            using (var deriveBytes = new PasswordDeriveBytes(TestPassword, s_testSalt))
+            {
+                Assert.Throws<CryptographicException>(() => deriveBytes.CryptDeriveKey("RC2", "SHA1", 128, null));
+                Assert.Throws<CryptographicException>(() => deriveBytes.CryptDeriveKey("RC2", "SHA1", 128, new byte[1]));
+            }
+        }
+
+        private static byte[] TestKnownValue_CryptDeriveKey(string password, string alg, string hashAlg, int keySize, byte[] salt, byte[] expected)
+        {
+            byte[] output;
+            byte[] iv = new byte[8];
+
+            using (var deriveBytes = new PasswordDeriveBytes(password, salt))
+            {
+                output = deriveBytes.CryptDeriveKey(alg, hashAlg, keySize, iv);
+            }
+
+            Assert.Equal(expected, output);
+
+            // For these tests, the returned IV is always zero
+            Assert.Equal(new byte[8], iv);
+
+            return output;
+        }
+
+        private static void TestKnownValue_GetBytes(string password, byte[] salt, int iterationCount, byte[] expected)
+        {
+            byte[] output;
+
+            using (var deriveBytes = new PasswordDeriveBytes(password, salt, "SHA1", iterationCount))
```

Comment:
I see MD5 and SHA256 were added to CryptDeriveKey.  This is for the PBKDF1-ish GetBytes... if we support other algorithms there we should validate that replacing the SHA1 identifier works.